woocommerce/woocommerce-gutenberg-products-block
# Prevent double triggering select shipping on the same rate

Shipping Rates components would trigger a redundant select call on mount, that call would also cause some data entered between the event start and end to be lost.

This PR prevents the redundant select from firing and would return early if the rate you're trying to select is alreay selected.

This does not fix the issue of the useEffect firing twice, unfrountantly I went into a deep rabbit hole,[ with resulted in me removing the local state](https://github.com/woocommerce/woocommerce-blocks/pull/8497/files/152787f033288c9d78dc59f2b16720f49bf13048) and updating the data store directly which I believe is better ([as you shouldn't have a component be controlled and uncontrolled](https://beta.reactjs.org/learn/sharing-state-between-components#controlled-and-uncontrolled-components) at the same time). That was a much better solution but ended up complicating the reducer with the switch logic.

This is something that we might revisit again in the future if needed.


Fixes https://github.com/woocommerce/woocommerce-blocks/issues/8485


### Testing

1. Set your default country to something like US/CA
2. Enable Local Pickup and add 2 locations to it.
3. In your Shipping zones, set shipping for Greece
4. Open Checkout, it should load on Local Pickup selected.
5. Change to regular shipping
6. Change country to Greece and set a city
7. Your City value should persist
8. Switch back to local pickup, it should be selected.
9. Switch back to regular shipping, it should work.


### WooCommerce Visibility





### Changelog

> Fix an issue in which setting the city/state would not persist in the first time for certain customers.
